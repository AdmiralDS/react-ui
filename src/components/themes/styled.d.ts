import 'styled-components';
import type { Theme } from './index';

interface IPalette {
  main: string;
  contrastText: string;
}

declare module 'styled-components' {
  export interface DefaultTheme extends Theme {
    description?: string;
  }
}
