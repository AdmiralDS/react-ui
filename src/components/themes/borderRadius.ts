import type { Shape } from '#src/components/themes/common';
export const ALL_BORDER_RADIUS_VALUES = [
  'Border radius 0',
  'Border radius 2',
  'Border radius 4',
  'Border radius 6',
  'Border radius 8',
  'Border radius 10',
] as const;

export function smallGroupBorderRadius(shape: Shape): string {
  switch (shape.borderRadiusKind) {
    case 'Border radius 0':
      return '0';
    case 'Border radius 2':
      return '2px';
    case 'Border radius 4':
      return '4px';
    case 'Border radius 6':
      return '4px';
    case 'Border radius 8':
      return '4px';
    case 'Border radius 10':
      return '4px';
    default:
      return '4px';
  }
}

export function mediumGroupBorderRadius(shape: Shape): string {
  switch (shape.borderRadiusKind) {
    case 'Border radius 0':
      return '0';
    case 'Border radius 2':
      return '2px';
    case 'Border radius 4':
      return '4px';
    case 'Border radius 6':
      return '6px';
    case 'Border radius 8':
      return '8px';
    case 'Border radius 10':
      return '10px';
    default:
      return '4px';
  }
}

export function largeGroupBorderRadius(shape: Shape): string {
  switch (shape.borderRadiusKind) {
    case 'Border radius 0':
      return '0';
    case 'Border radius 2':
      return '4px';
    case 'Border radius 4':
      return '8px';
    case 'Border radius 6':
      return '12px';
    case 'Border radius 8':
      return '16px';
    case 'Border radius 10':
      return '20px';
    default:
      return '8px';
  }
}
