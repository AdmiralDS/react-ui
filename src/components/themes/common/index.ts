import { Color } from '#src/components/themes/common/color';
import { ZIndex } from '#src/components/themes/common/zIndex';
import baseStyled, { ThemedStyledInterface } from 'styled-components';

export * from './color';

export interface Shadow {
  ClickableActive: string;
  ClickableDefault: string;
  ClickableHover: string;
  NonClickable: string;
}

export interface Theme {
  color: Color;
  shadow: Shadow;
  zIndex: ZIndex;
  name: 'light' | 'dark';
}

export const styled = baseStyled as ThemedStyledInterface<Theme>;
