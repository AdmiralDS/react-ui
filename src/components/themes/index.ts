export * from './light';
export * from './dark';
export type { Theme } from './common';
