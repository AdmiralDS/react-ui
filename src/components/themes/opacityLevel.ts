export const OPACITY_LEVEL_1 = 0.02;
export const OPACITY_LEVEL_2 = 0.04;
export const OPACITY_LEVEL_3 = 0.08;
export const OPACITY_LEVEL_4 = 0.1;
export const OPACITY_LEVEL_5 = 0.15;
export const OPACITY_LEVEL_6 = 0.16;
export const OPACITY_LEVEL_7 = 0.2;
export const OPACITY_LEVEL_8 = 0.25;
export const OPACITY_LEVEL_9 = 0.3;
export const OPACITY_LEVEL_10 = 0.35;
export const OPACITY_LEVEL_11 = 0.4;
export const OPACITY_LEVEL_12 = 0.5;
export const OPACITY_LEVEL_13 = 0.6;
export const OPACITY_LEVEL_14 = 0.8;
export const OPACITY_LEVEL_15 = 0.95;
export const OPACITY_MAX_LEVEL = 1;
