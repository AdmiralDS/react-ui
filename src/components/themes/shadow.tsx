import { css } from 'styled-components';

export const shadowClickableActiveMixin = css`
  box-shadow: 0 2px 4px rgba(138, 150, 168, 0.4), 0px -2px 2px rgba(138, 150, 168, 0.08);
`;

export const shadowClickableDefaultMixin = css`
  box-shadow: 0 4px 8px rgba(138, 150, 168, 0.4), 0px -2px 4px rgba(138, 150, 168, 0.08);
`;

export const shadowNonClickableMixin = css`
  box-shadow: 0 4px 16px rgba(138, 150, 168, 0.4), 0 0 6px rgba(138, 150, 168, 0.08);
`;

export const shadowClickableHoverMixin = css`
  box-shadow: 0 8px 16px rgba(138, 150, 168, 0.4), 0px -2px 8px rgba(138, 150, 168, 0.08);
`;
