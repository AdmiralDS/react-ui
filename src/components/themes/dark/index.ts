import type { Theme } from '../common';
import { COLOR as color } from './color';
import { shadow } from './shadow';
import { zIndex } from '../common/zIndex';

export const DARK_THEME: Theme = {
  color,
  shadow,
  zIndex,
  name: 'dark',
};
