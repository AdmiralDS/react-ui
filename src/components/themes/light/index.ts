import type { Theme } from '../common';
import { COLOR as color } from './color';
import { shadow } from './shadow';
import { zIndex } from '../common/zIndex';

export const LIGHT_THEME = {
  color,
  shadow,
  zIndex,
  name: 'light',
} as Theme;
