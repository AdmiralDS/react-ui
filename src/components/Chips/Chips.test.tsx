import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import userEvent from '@testing-library/user-event';
import { LIGHT_THEME } from '../themes';

import { Chips } from './';

describe('Chips', () => {
  it('should render default component', () => {
    const wrapper = render(
      <ThemeProvider theme={LIGHT_THEME}>
        <Chips>test</Chips>
      </ThemeProvider>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with appearanceTypeFilled prop ', () => {
    const wrapper = render(
      <ThemeProvider theme={LIGHT_THEME}>
        <Chips appearance="filled">test</Chips>
      </ThemeProvider>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with selected prop ', () => {
    const wrapper = render(
      <ThemeProvider theme={LIGHT_THEME}>
        <Chips selected>test</Chips>
      </ThemeProvider>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should call onClick when user clicks on component', () => {
    const onClick = jest.fn();
    const wrapper = render(
      <ThemeProvider theme={LIGHT_THEME}>
        <Chips onClick={onClick}>test</Chips>
      </ThemeProvider>,
    );
    fireEvent.click(wrapper.getByText('test'));
    expect(onClick).toBeCalledTimes(1);
  });

  it('should focus on component when user on press key tab', () => {
    const wrapper = render(
      <ThemeProvider theme={LIGHT_THEME}>
        <Chips data-testid="test-element">test</Chips>
      </ThemeProvider>,
    );
    expect(document.body).toHaveFocus();
    userEvent.tab();
    expect(wrapper.getByTestId('test-element')).toHaveFocus();
  });
});
