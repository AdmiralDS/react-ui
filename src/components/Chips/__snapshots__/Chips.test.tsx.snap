// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chips should render component with appearanceTypeFilled prop  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  position: relative;
  max-width: 190px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  pointer-events: auto;
  cursor: default;
  background-color: #E2E5E9;
  border: 1px solid transparent;
  border-radius: 16px;
  height: 32px;
  padding: 6px 12px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
}

.c0 *[fill^='#'] {
  fill: #626F84;
}

.c0:hover,
.c0:active {
  color: #2B313B;
  background-color: #D3D8DE;
}

.c0:focus-visible {
  outline: 0;
}

.c0:focus-visible:before {
  border: 2px solid #0A54C2;
  border-radius: 20px;
  content: '';
  display: block;
  position: absolute;
  top: -5px;
  left: -5px;
  right: -5px;
  bottom: -5px;
  pointer-events: none;
}

.c1 {
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 20px;
}

.c1 svg {
  height: 20px;
  width: 20px;
}

.c1 svg *[fill^='#'] {
  fill: #626F84;
}

.c2 {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  display: inline-block;
}

<body>
    <div>
      <div
        class="c0"
        tabindex="0"
      >
        <div
          class="c1"
        >
          <div
            class="c2"
          >
            test
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": .c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  position: relative;
  max-width: 190px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  pointer-events: auto;
  cursor: default;
  background-color: #E2E5E9;
  border: 1px solid transparent;
  border-radius: 16px;
  height: 32px;
  padding: 6px 12px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
}

.c0 *[fill^='#'] {
  fill: #626F84;
}

.c0:hover,
.c0:active {
  color: #2B313B;
  background-color: #D3D8DE;
}

.c0:focus-visible {
  outline: 0;
}

.c0:focus-visible:before {
  border: 2px solid #0A54C2;
  border-radius: 20px;
  content: '';
  display: block;
  position: absolute;
  top: -5px;
  left: -5px;
  right: -5px;
  bottom: -5px;
  pointer-events: none;
}

.c1 {
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 20px;
}

.c1 svg {
  height: 20px;
  width: 20px;
}

.c1 svg *[fill^='#'] {
  fill: #626F84;
}

.c2 {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  display: inline-block;
}

<div>
    <div
      class="c0"
      tabindex="0"
    >
      <div
        class="c1"
      >
        <div
          class="c2"
        >
          test
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Chips should render component with selected prop  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  position: relative;
  max-width: 190px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  pointer-events: auto;
  cursor: pointer;
  background-color: #0D69F2;
  border: 1px solid #0D69F2;
  border-radius: 16px;
  height: 32px;
  padding: 6px 12px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #FFFFFF;
}

.c0 *[fill^='#'] {
  fill: #0D69F2;
}

.c0:hover,
.c0:active {
  color: #FFFFFF;
  background-color: #0A54C2;
  border-color: #0A54C2;
}

.c0:focus-visible {
  outline: 0;
}

.c0:focus-visible:before {
  border: 2px solid #0A54C2;
  border-radius: 20px;
  content: '';
  display: block;
  position: absolute;
  top: -5px;
  left: -5px;
  right: -5px;
  bottom: -5px;
  pointer-events: none;
}

.c0:hover *[fill^='#'] {
  fill: #FFFFFF;
}

.c0:active *[fill^='#'] {
  fill: #0949AA;
}

.c1 {
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 20px;
}

.c1 svg {
  height: 20px;
  width: 20px;
}

.c1 svg *[fill^='#'] {
  fill: #FFFFFF;
}

.c2 {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  display: inline-block;
}

<body>
    <div>
      <div
        class="c0"
        tabindex="0"
      >
        <div
          class="c1"
        >
          <div
            class="c2"
          >
            test
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": .c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  position: relative;
  max-width: 190px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  pointer-events: auto;
  cursor: pointer;
  background-color: #0D69F2;
  border: 1px solid #0D69F2;
  border-radius: 16px;
  height: 32px;
  padding: 6px 12px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #FFFFFF;
}

.c0 *[fill^='#'] {
  fill: #0D69F2;
}

.c0:hover,
.c0:active {
  color: #FFFFFF;
  background-color: #0A54C2;
  border-color: #0A54C2;
}

.c0:focus-visible {
  outline: 0;
}

.c0:focus-visible:before {
  border: 2px solid #0A54C2;
  border-radius: 20px;
  content: '';
  display: block;
  position: absolute;
  top: -5px;
  left: -5px;
  right: -5px;
  bottom: -5px;
  pointer-events: none;
}

.c0:hover *[fill^='#'] {
  fill: #FFFFFF;
}

.c0:active *[fill^='#'] {
  fill: #0949AA;
}

.c1 {
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 20px;
}

.c1 svg {
  height: 20px;
  width: 20px;
}

.c1 svg *[fill^='#'] {
  fill: #FFFFFF;
}

.c2 {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  display: inline-block;
}

<div>
    <div
      class="c0"
      tabindex="0"
    >
      <div
        class="c1"
      >
        <div
          class="c2"
        >
          test
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Chips should render default component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  position: relative;
  max-width: 190px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  pointer-events: auto;
  cursor: default;
  background-color: #FFFFFF;
  border: 1px solid #0D69F2;
  border-radius: 16px;
  height: 32px;
  padding: 6px 12px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #0D69F2;
}

.c0 *[fill^='#'] {
  fill: #0D69F2;
}

.c0:hover,
.c0:active {
  color: #0A54C2;
  background-color: #F1F2F4;
  border-color: #0A54C2;
}

.c0:focus-visible {
  outline: 0;
}

.c0:focus-visible:before {
  border: 2px solid #0A54C2;
  border-radius: 20px;
  content: '';
  display: block;
  position: absolute;
  top: -5px;
  left: -5px;
  right: -5px;
  bottom: -5px;
  pointer-events: none;
}

.c0:hover *[fill^='#'] {
  fill: #0A54C2;
}

.c0:active *[fill^='#'] {
  fill: #0949AA;
}

.c1 {
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 20px;
}

.c1 svg {
  height: 20px;
  width: 20px;
}

.c1 svg *[fill^='#'] {
  fill: #0D69F2;
}

.c2 {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  display: inline-block;
}

<body>
    <div>
      <div
        class="c0"
        tabindex="0"
      >
        <div
          class="c1"
        >
          <div
            class="c2"
          >
            test
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": .c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  position: relative;
  max-width: 190px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  pointer-events: auto;
  cursor: default;
  background-color: #FFFFFF;
  border: 1px solid #0D69F2;
  border-radius: 16px;
  height: 32px;
  padding: 6px 12px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #0D69F2;
}

.c0 *[fill^='#'] {
  fill: #0D69F2;
}

.c0:hover,
.c0:active {
  color: #0A54C2;
  background-color: #F1F2F4;
  border-color: #0A54C2;
}

.c0:focus-visible {
  outline: 0;
}

.c0:focus-visible:before {
  border: 2px solid #0A54C2;
  border-radius: 20px;
  content: '';
  display: block;
  position: absolute;
  top: -5px;
  left: -5px;
  right: -5px;
  bottom: -5px;
  pointer-events: none;
}

.c0:hover *[fill^='#'] {
  fill: #0A54C2;
}

.c0:active *[fill^='#'] {
  fill: #0949AA;
}

.c1 {
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 20px;
}

.c1 svg {
  height: 20px;
  width: 20px;
}

.c1 svg *[fill^='#'] {
  fill: #0D69F2;
}

.c2 {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  display: inline-block;
}

<div>
    <div
      class="c0"
      tabindex="0"
    >
      <div
        class="c1"
      >
        <div
          class="c2"
        >
          test
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
